packages/create-plugin/src/migrations/utils.ts

```diff
@@ -132,3 +133,174 @@ export function installNPMDependencies(context: Context) {
+  const sortedDeps = sortObjectByKeys(currentDeps);
+  const sortedDevDeps = sortObjectByKeys(currentDevDeps);
+
+  // Only update if there are actual changes
+  const hasChanges =
+    JSON.stringify(sortedDeps) !== JSON.stringify(currentPackageJson.dependencies || {}) ||
+    JSON.stringify(sortedDevDeps) !== JSON.stringify(currentPackageJson.devDependencies || {});
+
+  if (!hasChanges) {
+    return;
+  }
+
+  const updatedPackageJson = {
+    ...currentPackageJson,
+    dependencies: sortedDeps,
+    devDependencies: sortedDevDeps,
+  };
+
+  context.updateFile(packageJsonPath, JSON.stringify(updatedPackageJson, null, 2));
+}
+
+export function removeDependenciesFromPackageJson(
+  context: Context,
+  dependencies: string[],
+  devDependencies: string[] = [],
+  packageJsonPath = 'package.json'
+) {
+  const currentPackageJson = readJsonFile(context, packageJsonPath);
+  const currentDeps = { ...(currentPackageJson.dependencies || {}) };
+  const currentDevDeps = { ...(currentPackageJson.devDependencies || {}) };
+
+  let hasChanges = false;
+
+  // Remove dependencies from dependencies section
+  for (const dep of dependencies) {
+    if (currentDeps[dep]) {
+      delete currentDeps[dep];
+      hasChanges = true;
+    }
+  }
+
+  // Remove dependencies from devDependencies section
+  for (const dep of devDependencies) {
+    if (currentDevDeps[dep]) {
+      delete currentDevDeps[dep];
+      hasChanges = true;
+    }
+  }
+
+  // Only update if there are actual changes
+  if (!hasChanges) {
+    return;
+  }
+
+  const updatedPackageJson = {
+    ...currentPackageJson,
+    dependencies: currentDeps,
+    devDependencies: currentDevDeps,
+  };
+
+  context.updateFile(packageJsonPath, JSON.stringify(updatedPackageJson, null, 2));
+}
+
+const UNIDENTIFIED_VERSION = 'UNIDENTIFIED_VERSION';
+// Handle special version strings like "latest", "next", etc.
+const DIST_TAGS = {
+  '*': 2,
+  UNIDENTIFIED_VERSION: 2,
+  next: 1,
+  latest: 0,
+};
+
+/**
+ * Compares two version strings to determine if the incoming version is greater
+ */
+function isIncomingVersionGreater(incomingVersion: string, existingVersion: string): boolean {
+  // if version is in the format of "latest", "next" or similar - keep it, otherwise try to parse it
+  const incomingVersionCompareBy =
+    incomingVersion in DIST_TAGS ? incomingVersion : (cleanSemver(incomingVersion)?.toString() ?? UNIDENTIFIED_VERSION);
+  const existingVersionCompareBy =
+    existingVersion in DIST_TAGS ? existingVersion : (cleanSemver(existingVersion)?.toString() ?? UNIDENTIFIED_VERSION);
+
+  if (incomingVersionCompareBy in DIST_TAGS && existingVersionCompareBy in DIST_TAGS) {
+    return (
+      DIST_TAGS[incomingVersionCompareBy as keyof typeof DIST_TAGS] >
+      DIST_TAGS[existingVersionCompareBy as keyof typeof DIST_TAGS]
+    );
+  }
+
+  if (incomingVersionCompareBy in DIST_TAGS || existingVersionCompareBy in DIST_TAGS) {
+    return true;
+  }
```

Comment:
Is this correct? 🤔 What if the latest version is `12.0.0`, the incoming version is `10.0.0`, while the existing version - which is also in dist tags - is `latest`?

Reply:
Yeah, I think it's good 👍